### ISSUE TITLE
OauthHook not passing in oauth_* to Request in auth_header=False

### ISSUE THREAD

**Issue by ghost** [maraujop/requests-oauth#20]
We're writing a nostest framework against our web service API using requests and requests-oauth where our API

In the setUp():

``` python
...
        if self.auth == "oauth":
            oauth_hook = OAuthHook(self.access_token, self.access_token_secret, 
                                   self.consumer_key, self.consumer_secret, 
                                   header_auth=False)
            self.client = requests.session(hooks={'pre_request': oauth_hook})
        else:
            self.client = requests
...
```

and then the method that makes the API webservice call:

``` python
...
            # Handle POST vs. GET
            if method == 'POST':
                data = self.append_auth(data=data) # doesn't do anything if 'oauth'
                req = self.client.post(url, data=data, allow_redirects=True)
            else:
                url = self.append_auth(url=url) # doesn't do anything if 'oauth'
                req = self.client.get(url, allow_redirects=True)
...
```

Our API authenticates the oauth_\* parameters through the HTTP request however when I output the API result, it's not showing the oauth_\* parameters passed through.

``` python
print "req.request=", req.request.__dict__
```

Note that when I do the header_auth=True, I do see the headers {Authorization...} int the request.

i.e,
I'm testing http://my.webservice.com/get?foo=bar

I would expect your oauthhook to do the following:

http://my.webservice.com/get?foo=bar&oauth_token=xxx&oauth_signature=xxx&oauth_nonce=xxx...etc...

But that's not what I'm seeing.  Am I missing something?

**Comment by ghost** [comment]
More debugging:

When I output the request object from OauthHook.**call**() 

``` javascript
'data_and_params': {'oauth_nonce': '42860964', 'oauth_timestamp': '1331831929', 'event_title': 'Test Event 7200', 'oauth_signature_method': 'HMAC-SHA1', 'event_type_id_list': '1,2', 'oauth_version': '1.0', 'publish_flag': 1, 'event_country': 'us', 'event_brief_description': 'Automatically Generated Test Event', 'oauth_signature': 'rEXw9NTLN/eyq7IHwZ90xkQmeB0=', 'event_city': 'austin', 'oauth_consumer_key': 'b513571b80685c74fa95b7755d916c24048889df4', 'event_creator_type': '1', 'oauth_token': '3183bf40ca1d0e7165127f8076bee5d404888f58d', 'event_state': 'tx', 'event_start_date': '2001-01-01'}
```

However the webservice is still seeing only:

```
Array\n(\n    [o] => json\n    [event_title] => Test Event 7200\n    [event_start_date] => 2001-01-01\n    [publish_flag] => 1\n    [event_creator_type] => 1\n    [event_type_id_list] => 1,2\n    [event_country] => us\n    [event_state] => tx\n    [event_brief_description] => Automatically Generated Test Event\n    [event_city] => austin\n)
```

I would expect to see [oauth_*] => '...' variables passed through.

Am I doing something wrong?

Cheers,

-Kevin

**Comment by maraujop** [comment]
Hi @mapmyfitness ,

Could you please let me know what version are you using ? 

At the moment I'm working on `dev` branch. I haven't release nothing stable, due to a bug in requests that is not fixed yet. You can read more on this here:
https://github.com/maraujop/requests-oauth/issues/17

I recommend you try `dev` branch with my fork of requests and see if you can reproduce the problem.

Thanks, cheers
Miguel

**Comment by ghost** [comment]
requests-oauth==0.3.0 is the version in use.  I'll give dev a whirl and respond in a few.  Thanks Miguel.

**Comment by ghost** [comment]
```
$ pip freeze
...
-e git://github.com/maraujop/requests-oauth.git@1223593fe3981076e04b67370729de58d9defe73#egg=requests_oauth-dev
....
```

The issue is still present.  I'll experiment with setting Content-Type to see if that forces thae data_params to be passed correctly as GET or POST variables.

**Comment by maraujop** [comment]
Couple more things.

If you try `dev`, don't forget you need my fork of requests so that it works. 

Re-reading your issue. I should say that oauth parameters not always go in the url when `header_auth` is set to `False`. On `GET` calls, they always do, while in `POST` requests sometimes they go in the url and sometimes they go in the body. It would be good if you could sniff the traffic that comes out of the client, to make sure your server is not popping anything when handling OAuth requests and you are looking at a place, where oauth stuff has been removed from the request anymore.

Cheers,
Miguel

**Comment by ghost** [comment]
```
$ pip freeze
...
-e git://github.com/maraujop/requests@42392c92fadc518b6a1387c2f5375d72cea13f28#egg=requests-dev
-e git://github.com/maraujop/requests-oauth.git@1223593fe3981076e04b67370729de58d9defe73#egg=requests_oauth-dev
...
```

That indeed did work. Thank you.  I'll keep an eye on this pull request.

-Kevin

**Comment by maraujop** [comment]
Great, I'm glad it works :)

**Comment by ghost** [comment]
Not a problem -- turned out requests-oauth has a fork that fixes this issue.  I'm currently using their dev branch of requests-oauth and their fork of your requests and the matter works.  I've cc'd Miguel to shed some light if that fork will become a pull.  Miguel: can you provide us a quick update?

https://github.com/maraujop/requests-oauth/issues/20

Thanks All

-Kevin

From: Kenneth Reitz <me@kennethreitz.com<mailto:me@kennethreitz.com>>
Date: Sat, 31 Mar 2012 01:07:31 -0500
To: Kevin Callahan <kevin@mapmyfitness.com<mailto:kevin@mapmyfitness.com>>
Cc: "_@kennethreitz.com<mailto:_@kennethreitz.com>" <_@kennethreitz.com<mailto:_@kennethreitz.com>>, Jeremy Boyd <jeremy.boyd@mapmyfitness.com<mailto:jeremy.boyd@mapmyfitness.com>>
Subject: Re: requests hook question

Hey Kevin,

Really sorry about the delayed response! I'm really behind on my email lately.

The issue looks resolved now  is this still a problem?

## 

Kenneth Reitz

On Friday, March 16, 2012 at 3:23 PM, Kevin Callahan wrote:

Hi Kenneth,

I met you out here in Austin when you presented Horuku as APUG.  We also started using your requests lib pretty heavily here which is snazzy.

I have an issue posted with requests-oauth I'm I'm wondering if you know of anything that could be at fault with the hook functionality on your end.
https://github.com/maraujop/requests-oauth/issues/20

Or I'm wondering if I'm looking at your hook methodology the wrong way?

Cheers,

-kevin

Kevin Callahan | Co-Founder, VP Innovation Strategy
kevin@mapmyfitness.commailto:kevin@mapmyfitness.com

MapMyFITNESS  Empowering Active Lifestyles!
http://www.mapmyfitness.comhttp://www.mapmyfitness.com/
1810 Blake Street  Denver, CO 80202
|C| 858.731.6051  |F| 877.580.5129  |Skype| mapmyrun

Home of MapMyRUN and MapMyRIDE

[cid:3BAE4655-8E4C-48CD-8526-F558368A58D3]

**Comment by maraujop** [comment]
Hi @mapmyfitness ,

I submitted a pull request to requests more than two months ago. You can see it here:
https://github.com/kennethreitz/requests/issues/445

There is actually another patch submitted for the same issue, that hasn't gone in and would be way easier to integrate. All I can say is that I've tried by all means to get any of the two merged in.

I will release a new stable version requiring a specific version of requests in the future, if this happens to be fixed.

Cheers,
Miguel

**Comment by maraujop** [comment]
Version 0.4.0 has been released, it works well with requests 0.12.1 or higher. I'm closing this issue.

Cheers
